### DISCUSSION TITLE
syslog device host names

### DISCUSSION THREAD

**Discussion by milesunder** [Security-Onion-Solutions/securityonion#6234] [Unsupported Versions] [answered]
Recently I've notice that the host names of the syslog devices forwarding to Security Onion via the forward node list the forward node in host.name field I want to have my syslog devices host name captured somewhere. 

After finding the following grok pattern https://github.com/Security-Onion-Solutions/securityonion/blob/afed0b70eb547081d533682c9078e2c728d6da29/salt/elasticsearch/files/ingest/syslog 

```bash
^<%{INT:syslog.priority}>%{DATA:syslog.timestamp} %{WORD:source.application}(\\[%{DATA:pid}\\])?: %{GREEDYDATA:real_message}$",
"^%{SYSLOGTIMESTAMP:syslog.timestamp} %{SYSLOGHOST:syslog.host} %{SYSLOGPROG:syslog.program}: CEF:0\\|%{DATA:vendor}\\|%{DATA:product}\\|%{GREEDYDATA:message2}$"
```
It seems the grok pattern should be using the SYSLOGHOST syntax as it's including it in the syslog timestamp. We are using the example log below.

````bash
<174>Nov 17 05:04:48 iDRAC-ASD-123 Severity: Informational, Category: Audit, MessageID: USR0031, Message: Unable to log in for root from 100.0.X.X using SSH.
```
Kibana displays information under syslog.timestamp registered multiple fields, includes the device name.
* syslog.timestamp.keyword: Nov 17 05:04:48 iDRAC-ASD-123
* syslog.timestamp.security Nov 17 05:04:48 iDRAC-ASD-123

Have I missed something or has this been done by design?

**Comment by abesinger** [comment, marked as answer]
It's not by design, the stock syslog pipeline isn't well though out. See #5251 for an example of an improved pipeline that at least does a better job of handling the default rsyslog format that every Linux distro (at least all the ones that come with rsyslog) uses.
That may or may not work for you out of the box.

I'm working towards putting that into a PR.

Regarding the hostname being the forwarding host instead of the originating host, I'm seeing this problem with various forwarders including filebeat.  I noted it in #5251 and also commented on it in #5553.  It's pretty amazing that the people developing log collectors don't seem to think it important to know which host the message came from, especially if the log message itself doesn't include the hostname (and there are numerous situations where that happens).
